app-experimental.tex

```diff
@@ -1311,6 +1315,69 @@ \subsection{Points-to-PCC} % <<<
+micro-operations in microarchitectures. Thus we propose to have the two operations
+of the complex \insnnoref{CInvokeInd} to be split into a linked pair of instructions:
+an instruction for load of the code capability and an instruction for jumping to the
+code capability and unsealing the data capability.
+This property of having linked will retain the atomicity of the both IDC and PCC
+being written with valid capabilities or none of them being written.
+
+\subsubsection{Definition of linked pair of instructions}
+
+In this extension, we propose to split the cisentryj (CHERI indirect sealed entry jump) instruction into two sub instructions: Lil (load isentry linked) and cjaurl (CHERI jump and unseal into register linked).
+In the following paragraphs, we will explain what these instructions do and how they enforce the security model we are designing.
+
+The lil instruction loads a capability through an isentry capability and clears its tag.
+In the subsequent instruction this capability can be picked up by a cjaurl, which then jumps to this code capability and unseals the isentry into ct6.
+The two instructions are linked and all security properties are held as explained in the following paragraphs.
+
+If attackers executed only the lil instruction, this will leave them with an untagged code capability.
+This code capability cannot be used in regular jumps because it is untagged.
+The only way it can be used for a jump is in a subsequent cjaurl instruction.
+
+An attacker can only jump to a code capability with the isentry it came from.
+This can be divided into two cases: If the instruction following the lil is a cjaurl, this will jump to the code pointed to by the code capability and unseal the isentry.
+The register holding the isentry is stored in the reservation information of the lil instruction.
+In the second case, if the instruction following the lil is not a cjaurl, the reservation will have become invalid.
+Thus the attacker cannot jump to the code capability.
+Equally, the isentry has never been unsealed and thus cannot be used by the attacker.
+
+Because of this linkage, the attacker can never use a code capability and a data capability together, which have not been intended to work together.
+Thus the linking keeps the atomicity of the instruction.
+
+Loading the code capability does not break CHERI confidentiality.
+Isentries are not designed for secrecy about where a compartment wants to jump, but to implement jumping to a sealed pair of capabilities.
+Similar methods, e.g., cinvoke gives out sealed code and data capabilities, which are then invoked as a pair.
+The code capability itself is not secret.
+
+It is important to note that the linked pair of a lil and cjaurl instruction does not give any guarantees about memory consistency. Memory consistency is not an issue tackled by CHERI, but needs to be addressed by other architectural mechanisms, e.g. atomic memory operations.
```

Comment:
The reason for this paragraph to exist is to avoid confusing people who think this `ll` - `sc` like concept. I agree the consequences could be interesting. Let's have a chat about that later. For now, I will include your changes:)